saleor/graphql/discount/types/sales.py

```diff
@@ -121,81 +112,101 @@ class Meta:
-        root: ChannelContext[models.Sale], info: ResolveInfo, **kwargs
+        root: ChannelContext[models.Promotion], info: ResolveInfo, **kwargs
     ):
-        qs = root.node.categories.all()
-        return create_connection_slice(qs, info, kwargs, CategoryCountableConnection)
+        from ..utils import convert_migrated_sale_catalogue_predicate
+
+        if predicates := convert_migrated_sale_catalogue_predicate(root.node):
+            if category_ids := predicates.get("categoryPredicate"):
+                qs = Category.objects.filter(id__in=category_ids)
+                return create_connection_slice(
+                    qs, info, kwargs, CategoryCountableConnection
+                )
 
     @staticmethod
-    def resolve_channel_listings(root: ChannelContext[models.Sale], info: ResolveInfo):
-        return SaleChannelListingBySaleIdLoader(info.context).load(root.node.id)
+    def resolve_channel_listings(
+        root: ChannelContext[models.Promotion], info: ResolveInfo
+    ):
+        if rule := root.node.rules.first():
+            if channel := rule.channels.first():
+                return [
+                    SaleChannelListing(
+                        # TODO what about id???
+                        id=graphene.Node.to_global_id("SaleChannelListing", rule.id),
+                        channel=channel,
+                        discount_value=rule.reward_value,
+                        currency=channel.currency_code,
+                    )
+                ]
 
     @staticmethod
     def resolve_collections(
-        root: ChannelContext[models.Sale], info: ResolveInfo, **kwargs
+        root: ChannelContext[models.Promotion], info: ResolveInfo, **kwargs
     ):
-        qs = root.node.collections.all()
-        qs = ChannelQsContext(qs=qs, channel_slug=root.channel_slug)
-        return create_connection_slice(qs, info, kwargs, CollectionCountableConnection)
+        from ..utils import convert_migrated_sale_catalogue_predicate
+
+        if predicates := convert_migrated_sale_catalogue_predicate(root.node):
+            if collection_ids := predicates.get("collectionPredicate"):
+                qs = Collection.objects.filter(id__in=collection_ids)
+                qs = ChannelQsContext(qs=qs, channel_slug=root.channel_slug)
+                return create_connection_slice(
+                    qs, info, kwargs, CollectionCountableConnection
+                )
 
     @staticmethod
     def resolve_products(
-        root: ChannelContext[models.Sale], info: ResolveInfo, **kwargs
+        root: ChannelContext[models.Promotion], info: ResolveInfo, **kwargs
     ):
-        qs = root.node.products.all()
-        qs = ChannelQsContext(qs=qs, channel_slug=root.channel_slug)
-        return create_connection_slice(qs, info, kwargs, ProductCountableConnection)
+        from ..utils import convert_migrated_sale_catalogue_predicate
+
+        if predicates := convert_migrated_sale_catalogue_predicate(root.node):
```

Comment:
This function causes additional database queries. We should fetch required data via dataloader and pass it to any required functions inside resolvers.